app/script/auth/module/action/SelfAction.js

```diff
@@ -32,6 +33,23 @@ export function fetchSelf() {
         });
       })
       .then(selfUser => dispatch(SelfActionCreator.successfulFetchSelf(selfUser)))
-      .catch(error => dispatch(SelfActionCreator.failedFetchSelf(error)));
+      .catch(error => {
+        dispatch(SelfActionCreator.failedFetchSelf(error));
+        throw BackendError.handle(error);
+      });
+  };
+}
+
+export function setHandle(handle) {
+  return function(dispatch, getState, {apiClient}) {
+    dispatch(SelfActionCreator.startSetHandle());
+    return apiClient.self.api
+      .putHandle({handle: handle.toLowerCase()})
```

Comment:
Is that all the normalization we need? What about trimming to remove whitespaces?

Reply:
if more complicated maybe have a normalisation function for the `handle`?